### PR TITLE
Split tokio feature into tokio and tokio-tungstenite

### PR DESCRIPTION
**Pull request by teohhanhui**

### CODE REVIEW

File: tests/interop/main.rs [hunk 1]
```diff
@@ -123,7 +123,7 @@ fn start_js_server() -> Child {
 
 fn npm_build() {
     println!("npm run build");
-    let mut cmd = std::process::Command::new("/usr/bin/npm");
+    let mut cmd = std::process::Command::new("npm");
```
Comment (teohhanhui) [inline tests/interop/main.rs:126]: Hardcoding npm path doesn't work in many cases. For example when using https://github.com/Schniz/fnm

**Comment by gterzian** [comment]
LGTM, @alexjg ?

**Comment by alexjg** [comment]
:+1:

File: tests/interop/main.rs (hunk 1 above)
Comment (alexjg) [reply tests/interop/main.rs:126]: Yeah good spot, I fixed this in #29

**Comment by teohhanhui** [comment]
Rebased.

File: Cargo.toml [hunk 2]
```diff
@@ -24,7 +23,7 @@ crossbeam-channel = { version = "0.5.8" }
 parking_lot = { version = "0.12.1" }
 futures = { version = "0.3.28" }
 serde = { version = "1.0.160", features = ["derive"]}
-tokio = { version = "1.27", features = ["full"], optional = true }
+tokio = { version = "1.27", features = ["rt"], optional = true }
```
Comment (teohhanhui) [inline Cargo.toml:26]: https://docs.rs/tokio/latest/tokio/#authoring-libraries

**Comment by teohhanhui** [comment]
No idea what's up with the CI job: https://github.com/automerge/spanreed/actions/runs/5556816827/jobs/10149810872?pr=34

**Comment by alexjg** [comment]
Yeah that's #36

**Pull request merged**